File: oddoccur.py
def check_odd_occurences(data_array):
    for i, val in enumerate(data_array):
        if (data_array.count(val) % 2) == 1:
            return i
    
    return -1;

data = [9, 3, 9, 3, 9, 7, 9];
jomblo = check_odd_occurences(data)
print("elemen pada indeks ke-", jomblo, " (",data[jomblo],") tidak mempunyai pasangan")

File: oddoccur.java
/**
 * oddoccur
 */
public class oddoccur {
    public static void main(String[] args) {
        int[] data = { 9, 3, 9, 3, 9, 7, 9 };
        int jomblo = checkOddOccurences(data);
        System.out.println("element ke-" + jomblo + "(" + data[jomblo] + ") tidak mempunyai pasangan");
    }   
    public static int checkOddOccurences(int[] dataArray) {
        for (int i=0; i<dataArray.length; i++) {
            int num =0;
            for (int j=0; j<dataArray.length; j++){
                if (dataArray[i] == dataArray[j]) {
                    num++;
                }
            }
            if (num % 2 == 1){
                return i;
            }
        }
        return -1;
    } 
}